pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -966,6 +977,45 @@ func (ctrl *PersistentVolumeController) assignDefaultStorageClass(ctx context.Co
 	return true, nil
 }
 
+// assignDefaultVolumeAttributesClass updates the claim volume attributes class if there is any, the claim is updated to the API server.
+// Ignores claims that already have a volume attributes class.
+// TODO: if resync is ever changed to a larger period, we might need to change how we set the default volume attributes class on existing claims
+func (ctrl *PersistentVolumeController) assignDefaultVolumeAttributesClass(ctx context.Context, claim *v1.PersistentVolumeClaim) (bool, error) {
+	logger := klog.FromContext(ctx)
+
+	if claim.Spec.VolumeAttributesClassName != nil {
+		return false, nil
+	}
+
+	className := storagehelpers.GetPersistentVolumeClaimClass(claim)
+	if className == "" {
+		return false, nil
+	}
+	sc, err := ctrl.classLister.Get(className)
+	if err != nil {
+		return false, err
+	}
+
+	vac, err := util.GetDefaultVolumeAttributesClass(ctrl.vacLister, sc.Provisioner)
+	if err != nil {
+		return false, err
+	} else if vac == nil {
+		logger.V(4).Info("Can not assign volume attributes class to PersistentVolumeClaim: default volume attributes class not found", "driverName", sc.Provisioner, "PVC", klog.KObj(claim))
```

Comment:
This is a valid and common scenario. This log level should be dropped signficantly more, like at least 5.